mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMessageManager.java

```diff
@@ -410,6 +421,40 @@ public Long execute() throws MailboxException {
 
     }
 
+    protected void addAttachmentstoStore(Iterable<Content> attachments, MailboxSession mailboxSession) {
+        AttachmentMapper attachmentMapper = getAttachmentMapper(mailboxSession);
+        for (Content attachment : attachments) {
+            Attachment newAttachment = new SimpleAttachment(AttachmentBlobId.getNewId(), attachment);
+            attachmentMapper.put(newAttachment);
+        }
+    }
+
+    private static class AttachmentBlobId implements AttachmentId {
+
+        private final UUID id;
+
+        public static AttachmentId getNewId() {
+            return new AttachmentBlobId(UUID.randomUUID());
+        }
+
+        public static AttachmentId of(String blobId) {
+            return new AttachmentBlobId(UUID.fromString(blobId));
+        }
+
+        private AttachmentBlobId(UUID id) {
+            this.id = id;
+        }
+
+        @Override
+        public String serialize() {
+            return id.toString();
+        }
```

Comment:
Won't you have troubles in your tests if you don't have equals and hashcode implemented ?